compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -1,7 +1,7 @@
 //! The virtual memory representation of the MIR interpreter.
 
-mod init_mask;
-mod provenance_map;
+pub mod init_mask;
+pub mod provenance_map;
```

Comment:
This was all quite deliberately sealed away. Do we really have to make this public? :(

Reply:
We're gonna make it private before this PR lands. We're not exposing the internals, but adding helper methods that avoid having to expose them.

Reply:
You don't need to add any helper methods. The existing methods are sufficient for the interpreter, codegen, and MiniRust.

Reply:
Yes, but these do random access, we want to duplicate the information. Iterating over all bytes is... not great.

Reply:
[Here is how MiniRust does it](https://github.com/RalfJung/minirust/blob/196da9640a4cd608405ad40110d29930827a99a5/tooling/minimize/src/constant.rs#L132-L145)

Reply:
Yea, I assumed that was too expensive for big allocations, but we can optimize when we see that being an issue

Reply:
Made them private, I think in future PR we should def get inspiration from MiniRust.